docs/manual/src/internals/object_references.md

```diff
@@ -10,21 +10,35 @@ including:
   active at any point in the program.
 * Guarding against data races.
 
-UniFFI aims to maintain these guarantees even when the Rust code is being invoked
-from a foreign language, at the cost of turning them into run-time checks rather
-than compile-time guarantees.
+The very nature of the problems UniFFI tries to solve is that calls may come
+from foreign languages on any thread, outside of the control of Rust's ownership
+system. UniFFI itself tries to take a hands-off approach as much as possible and
+depends on the Rust compiler itself to uphold safety guarantees, without assuming
+that foreign-language callers will be "well behaved".
```

Comment:
@mhammond FYI I've liberally appropriated the docs updates from your #420.

Reply:
great - so the nits below are probably mine :)